_posts/2021-04-21-frequency-band-control-in-unetstack.md

```diff
@@ -0,0 +1,199 @@
+
+The other parameter involved in bandwidth control is `phy[2].dc0`. This controls the starting carrier number. 
+
+`dc0 >= -nc/2 || dc0 <= nc/2`
+
+This can be used to shift the centre of the frequency band "left" or "right". 
+
+As mentioned above, to set the band to match the transducer, we need `C=26kHz` and `B=12kHz`.
+
+The default modem parameters are set as follows to achieve this.
+```
+> phy[2]
+  ...
+  bw = 0.5
+  dc0 = 85
+  nc = 1024
+```
+
+`phy.dc0=85` shifts the centre by approximately `2000Hz`. Carriers go from -512 to +512 to span the band 12kHz to 36kHz. `85/512 ~= 1/6`, so the centre will shift roughly by `1/6` of 12kHz = 2000Hz (the positive half band) to 26kHz from 24kHz.
+
+`phy[2].bw=0.5` will reduce the active carriers by half, so a bandwidth of 12kHz. Thus together, the used bandwidth will go from 20kHz to 32kHz.
+
+For Unet audio, the same settings will shift the centre by 1kHz to 13kHz with a bandwidth of 6kHz. Therefore from 10kHz to 16kHz, this is seen below (plvl=-25, trigger -70, may vary for user).
+
+<p align="center"><img src="../assets/img/freqBandControl/ofdm-psd.png" style="zoom:50%;"/></p>
+
+Note that if there is a frequency band to be avoided, we can position the used band to the "left" or "right" of it by adjusting `nc`, `dc0` and `bw`. Note that going far away from the ideal transducer resonance region will reduce transmission efficiency. 
+
+The usage and meaning of parameters can be found in [Command Reference](https://unetstack.net/handbook/unet-handbook.html#_command_reference)
+
+## FHBFSK frequency band control
+
+For FHBFSK scheme, the parameter `phy[3].fmin` specifies the starting frequency. `phy[3].fstep` specifies the frequency steps and  `phy[3].hops` the total number of frequencies used. 
+
+By default you may see in the Unet audio modem 
+```
+> phy[3]
+  ...
+  fmin = 9520.0
+  fstep = 160.0
+  hops = 13
+```
+
+So it starts from 9520 Hz and goes to 9520 + 160 x (13-1) x 2  = 13360 Hz as seen below. 
+The effective bandwidth can be roughly taken as fstep x hops x 2.
+
+
+<p align="center"><img src="../assets/img/freqBandControl/fhbfsk-psd.png" style="zoom:50%;"/></p>
+
+
+We can alter the above parameters to use a band anywhere between 6kHz and 18kHz, which is the computationally feasible band of the Unet audio modem. 
+
+Note that just as in OFDM, if there is a frequency band to be avoided, we can position the used band to the "left" or "right" of it by adjusting `fmin`, `fstep` and `hops`.
+
+
+## Control of Preamble bandwidth
+
+To complete the picture of bandwidth control, it must be noted that the signal that is transmitted not only has a data modulated portion, but also a detection preamble at the start. This is usually a pseudo random noise signal or m-sequences, hyperbolic chirp (HFM) etc that has good auto correlation properties. 
```

Comment:
Might want to give the default preambles for all schemes for UnetStack-based modems here.

Reply:
Can one of you add this?

Reply:
I looked for this in the unet handbook, but couldn't find it as it is modem related param. Do you mean to provide the actual preamble signal that we use @mchitre ?

Reply:
I mean the settings equivalent for the default preambles we use. They are in `physical.c`